sw/host/tests/chip/spi_passthru/src/main.rs

```diff
@@ -60,6 +64,71 @@ fn test_jedec_id(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
     Ok(())
 }
 
+fn test_enter_exit_4b_mode(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let spi = transport.spi(&opts.spi)?;
+
+    log::info!("Entering 4B address mode");
+    spi.run_transaction(&mut [Transfer::Write(&[SpiFlash::ENTER_4B])])?;
+    let sr = StatusRegister::read(&*uart)?;
+    assert!(sr.addr_4b, "expected to be in 4b mode");
+
+    log::info!("Exiting 4B address mode");
+    spi.run_transaction(&mut [Transfer::Write(&[SpiFlash::EXIT_4B])])?;
+    let sr = StatusRegister::read(&*uart)?;
+    assert!(!sr.addr_4b, "expected to be in 3b mode");
+    Ok(())
+}
+
+fn test_write_enable_disable(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let spi = transport.spi(&opts.spi)?;
+
+    log::info!("Sending WRITE_ENABLE");
+    spi.run_transaction(&mut [Transfer::Write(&[SpiFlash::WRITE_ENABLE])])?;
+    let status = SpiFlash::read_status(&*spi)?;
+    let sr = StatusRegister::read(&*uart)?;
+    assert!(
+        status as u32 & FLASH_STATUS_WEL != 0,
+        "expected WEL set via read_status"
+    );
+    assert!(
+        sr.status & FLASH_STATUS_WEL != 0,
+        "expected WEL set on the device"
+    );
+
+    log::info!("Sending WRITE_DISABLE");
+    spi.run_transaction(&mut [Transfer::Write(&[SpiFlash::WRITE_DISABLE])])?;
+    let status = SpiFlash::read_status(&*spi)?;
+    let sr = StatusRegister::read(&*uart)?;
+    assert!(
+        status as u32 & FLASH_STATUS_WEL == 0,
+        "expected WEL clear via read_status"
+    );
+    assert!(
+        sr.status & FLASH_STATUS_WEL == 0,
+        "expected WEL clear on the device"
+    );
+    Ok(())
+}
+
+fn test_read_status_extended(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let spi = transport.spi(&opts.spi)?;
+
+    let sr = StatusRegister {
+        status: 0x5A55AA,
+        addr_4b: false,
+    };
+    sr.write(&*uart)?;
+    // FIXME(cfrantz): Is this a bug?  We need a transaction to latch
+    // the staged flash_status into the register.
+    spi.run_transaction(&mut [Transfer::Write(&[SpiFlash::NOP])])?;
```

Comment:
A sharp edge? For initial software-provided values, the assumption is that Read ID or Read SFDP will execute first, allowing the staged value to commit.

If the status registers need to update independently of initial values or Write Status, then we have a bug. (WEL and WIP's special handling aside)

Reply:
I think the current behavior is acceptable.  I left a comment explaining the behavior and the reason for the NOP transaction.